### DISCUSSION TITLE
Does the phrase "Fix always available" in the docs suggest that it is a 'safe' fix for the `ruff check --fix-only` command?

### DISCUSSION THREAD

**Discussion by dimi-tree-v** [astral-sh/ruff#9832] [Q&A]
Taking the example of rule [RET504](https://docs.astral.sh/ruff/rules/unnecessary-assign/#:~:text=Fix%20is%20always%20available.), I would expect code breaking this rule to be fixed by the command:
```
ruff check --fix-only
```
as this rule is described as "Fix always available". However, instances of this issue were considered "unsafe", and not fixed.

It is my expectation that rules marked "Fix always available" are considered "safe" to auto-fix. Since this doesn't seem to be the case, can I please have clarification on what "Fix always available" means here?

**Comment by zanieb** [comment]
Hi! "Always available" means that whenever we raise the violation we can provide a fix — it may or may not be safe. For some violations we are only able to provide a fix in certain cases, which lists it as "sometimes available".

**Reply by allanlewis** [reply]
OK, so "fix always available" includes "unsafe" fixes? If so, that's fine but it would be good to clarify this somewhere in the docs.
